### DISCUSSION TITLE
请问关于输入系统为什么没有使用事件机制？

### DISCUSSION THREAD

**Discussion by rocketman123456** [BoomingTech/Piccolo#305] [Q&A] [answered]
```
unsigned int command = g_runtime_global_context.m_input_system->getGameCommand();
```
在引擎中，键盘按键仅通过input system来直接获取，并通过getCommand方法来取得实际输入，但是这样的设计耦合度比较高，会不会不利于后续扩展？后续会准备设计事件系统来替换这个部分吗？

**Comment by Ol6rin** [comment, marked as answer]
输入系统确实存在你说的问题。完备的输入系统应该复杂得多，利用事件机制只是一部分，实际按键需要经过改键映射变为虚拟按键，然后虚拟按键需要通过能够配置转化为基础游戏指令，基础游戏指令经过组合输出实际游戏指令，最终触发相应游戏逻辑。小引擎中目前输入系统是比较hack的简单实现，后续视开发计划情况再改进。
